api/v1alpha3/azuremachine_webhook.go

```diff
@@ -105,6 +105,10 @@ func (m *AzureMachine) ValidateUpdate(oldRaw runtime.Object) error {
 		allErrs = append(allErrs, errs...)
 	}
 
+	if errs := ValidateManagedDisks(old.Spec.DataDisks, m.Spec.DataDisks, field.NewPath("dataDisks")); len(errs) > 0 {
```

Comment:
Can we add a test for this?

Reply:
Also, there is already a `ValidateDataDisks` which validates `spec.DataDisks`. Maybe update that to include `ValidateManagedDisks` check?

Reply:
After trying out a few possibilities, I think it's best if we have this as a separate function as we are only validating managed disks for data disks. If we want to incorporate this in `ValidateDataDisks`, the function signature would change and would not be consistent with `ValidateOsDisk`. Any ideas/suggestions here?

Reply:
I thought it was slightly confusing to have two validation methods for the same field. I don't mind changing the signature if needed. If you think changes are going to be big, I'm okay with just adding comments for now, and doing it in a separate refactoring pr.

Reply:
Makes sense. I changed the function name to `ValidateDataDisksUpdate` to make it less confusing and added more checks. Thank you for your feedback.